src/java_bytecode/expr2java.h

```diff
@@ -58,7 +58,8 @@ std::string type2java(const typet &type, const namespacet &ns);
 template <typename float_type>
 std::string floating_point_to_java_string(float_type value)
 {
-  const auto is_float = std::is_same<float_type, float>::value;
+  static const auto is_float = std::is_same<float_type, float>::value;
+  (void)is_float; // silence a spurious warning generated by GCC 8
   static const std::string class_name = is_float ? "Float" : "Double";
```

Comment:
Uh, this one is sad

Reply:
Yes. I changed the name of the variables where it is being used and that does cause a compile failure. So it really seems to be a bug...